static/app/components/modals/dashboardWidgetLibraryModal/index.tsx

```diff
@@ -98,21 +100,36 @@ function DashboardWidgetLibraryModal({
             <SelectedBadge data-test-id="selected-badge">
               {`${selectedWidgets.length} Selected`}
             </SelectedBadge>
-            <Button
-              data-test-id="confirm-widgets"
-              priority="primary"
-              type="button"
-              onClick={(event: React.FormEvent) => {
-                event.preventDefault();
-                if (!!!selectedWidgets.length) {
-                  setErrored(true);
-                  return;
+            <Tooltip
+              title={tct(
+                'Exceeds max widgets ([maxWidgets]) per dashboard. Plese uselect [unselectWidgets] widget(s).',
+                {
+                  maxWidgets: MAX_WIDGETS,
+                  unselectWidgets:
+                    dashboard.widgets.length + selectedWidgets.length - MAX_WIDGETS,
                 }
-                handleSubmit();
-              }}
+              )}
+              disabled={
+                !!!(dashboard.widgets.length + selectedWidgets.length > MAX_WIDGETS)
```

Comment:
I'm wondering, can we pull `dashboard.widgets.length + selectedWidgets.length > MAX_WIDGETS` into a variable like `isOverLimit` (I'm not sure I like this name) and then this gets `disabled={!isOverLimit}` and the button is `disabled={isOverLimit}`

Reply:
09d83035b9a57fc6fbc9e04a0fff52f42370e721